File: src/main/kotlin/com/scottscmo/model/bible/BibleDB.java
package com.scottscmo.model.bible;

import com.scottscmo.model.Database;

import java.io.IOException;
import java.sql.Connection;
import java.sql.PreparedStatement;
import java.sql.SQLException;
import java.sql.Statement;
import java.util.function.Consumer;

final class BibleDB {
    private static final Database db;

    static {
        try {
            db = new Database("bible");
        } catch (IOException e) {
            throw new RuntimeException("Unable to find bible db", e);
        }
    }

    static Connection connect() throws SQLException {
        return db.connect();
    }

    static void useStatement(Consumer<Statement> run) throws SQLException {
        try (Statement stmt = connect().createStatement()) {
            run.accept(stmt);
        }
    }

    static void usePrepareStatement(String sql, Consumer<PreparedStatement> run) throws SQLException {
        try (PreparedStatement stmt = connect().prepareStatement(sql)) {
            run.accept(stmt);
        }
    }

    static boolean isEmpty() {
        return db.isEmpty();
    }
}

File: src/main/kotlin/com/scottscmo/model/bible/BibleDB.kt
package com.scottscmo.model.bible

import com.scottscmo.model.Database
import java.sql.Connection
import java.sql.PreparedStatement
import java.sql.SQLException
import java.sql.Statement

internal object BibleDB {
    private val db = Database("bible")

    @Throws(SQLException::class)
    fun connect(): Connection? {
        return db.connect()
    }

    fun useStatement(run: (stmt: Statement) -> Unit) {
        var stmt: Statement? = null
        try {
            stmt = connect()!!.createStatement()
            run(stmt)
        } finally {
            stmt?.close()
        }
    }

    fun usePrepareStatement(sql: String, run: (stmt: PreparedStatement) -> Unit) {
        var stmt: PreparedStatement? = null
        try {
            stmt = connect()!!.prepareStatement(sql)
            run(stmt)
        } finally {
            stmt?.close()
        }
    }

    fun isEmpty(): Boolean {
        return db.isEmpty
    }
}